src/core/math_func.hpp

```diff
@@ -361,6 +361,30 @@ static inline int DivAwayFromZero(int a, uint b)
 	}
 }
 
+/**
+ * Computes a / b rounded towards negative infinity for signed a and unsigned b.
```

Comment:
Documentation says "signed a and unsigned b", but function just takes the same type. Could be really fancy and use https://en.cppreference.com/w/cpp/types/is_unsigned with an `assert_compile` if you like ;)

Reply:
You know, cppreference.com has become my new favorite website over the last few weeks? It's brilliant!

Reply:
I decided to go for the best of both worlds; an `assert_compile` to guarantee we're dealing with arithmetic types, and a runtime `assert` if `unsigned b` turns out to be signed after all. If I can make the compiler spit out a warning that `b` should be an unsigned type, then that's all the better :wink:

Reply:
I also threw in a `static_cast` to the type of `a`, since that seems to be the in thing to do with these particular functions.

Reply:
Instead of "signed a and unsigned b" it should probably say something like "positive values of b" or "b > 0".
T does not strictly have to be signed for the function to return correct results.

Reply:
All the other related functions use an unsigned type for the divisor; and I have roofed the function to work either way; if it's fed s signed divisor, you'll get a run-time assertion, else a compile-time assertion.

Reply:
Is there any reason why these functions actually need to be templated? I have been toying with templating all the other related functions; and there's a lot to get right. I actually got my abomination to compile; but there's no way to guarantee the resulting program is correct 🙄.